src/components/Form/index.tsx

```diff
@@ -66,11 +66,25 @@ const Form: ComponentProps = ({
     ...rest
 }) => {
     const classes = useStyles();
-    // console.log(joiSchema.describe());
     const resolver = React.useMemo(
         () => joiResolver(validation || getValidation(schema), {stripUnknown: true, abortEarly: false}),
         [validation, schema]
     );
+    const validationSchema = React.useMemo(
+        () => validation || getValidation(schema),
+        [validation, schema]
+    );
+    const isPropertyRequired = React.useCallback((propertyName) => {
+        try {
+            const schema = validationSchema.extract(propertyName);
+            return schema._flags.presence === 'required';
```

Comment:
To avoid this dependency on joi internals, the required fields can be derived from `schema` variable instead. The `validationSchema` is anyway derived from there.